Poly/Poly/Poly.java

```diff
@@ -0,0 +1,113 @@
+package main;
+public class Poly 
+{
+    private final int coefficient[];
+    /**
+     * Parameterized Constructor used to initialize coefficient variable 
+     * @param inputCoefficient denotes the coefficients entered by the user 
+     */
+    public Poly(int inputCoefficient[])
+    {
+        this.coefficient=inputCoefficient;
+        if(validatePoly())
+            throw new AssertionError("Coefficient cannot be zero");
+    }
+    /**
+     * This method is used to check whether the coefficient entered by the user is zero or not
+     * @return true if coefficient is zero else return false
+     */
+    private boolean validatePoly()
+    {
+        boolean flag=true;
+        for(int i=0;i<coefficient.length;i++)
+        {
+            if(coefficient[i]==0)
+            {
+                flag=true;
+                break;
+            }
+            else 
+                flag=false;
+        }
+        return flag;
+    }
+    public int degree()
+    {
+        return coefficient.length-1;
+    }
+    /**
+     * Evaluation of the polynomial expression
+     * @param value the value of the variable
+     * @return the result of the evaluated polynomial
+     */
+    public float evaluate(float value)
+    {
+        float answer=0.0f;
+        int degreeOfPolynomial=degree();
+        for(int i=0;i<coefficient.length;i++)
+        {
+            answer+=coefficient[i]*Math.pow(value,degreeOfPolynomial);
+            degreeOfPolynomial--;
+        }
+        return answer;
+    }
+    /**
+     * Adding two polynomials
+     * @param p1 The first polynomial
+     * @param p2 the second polynomial to be multiplied
+     * @return new reference to the added polynomial
+     */
+    public int[] addPoly(Poly p1,Poly p2)
+    {
+        int poly1Coeffecient[] = p1.coefficient;

```

Comment:
You can optimise this more. I think it could be implemented with largePoly and smallPoly array only.

Reply:
Will discuss this